packages/devp2p/src/eth/index.ts

```diff
@@ -217,7 +217,10 @@ export class ETH extends EventEmitter {
         this._latestBlock = latestBlock
       }
       const forkHashB = Buffer.from(this._forkHash.substr(2), 'hex')
-      const nextForkB = this._nextForkBlock.toArrayLike(Buffer)
+      const nextForkB = this._nextForkBlock.eqn(0)
+        ? Buffer.from('', 'hex')
+        : this._nextForkBlock.toArrayLike(Buffer)
+
```

Comment:
Just to confirm here, so this only applies to genesis block comparisons, if I read the code correctly?

Reply:
No, this means that if we are on the final fork currently known (so on Aleut, at block 10 we switch to London, and then after that there is no hardfork), we report the empty buffer. If you instead do `this._nextForkBlock.toArrayLike(Buffer)`, then the reported Buffer is `Buffer.from('00', 'hex')`. If we send this over the wire, other nodes (at least Geth) will disconnect.

Reply:
Ah ok, thanks for clarification. We had this initially wrong along receiving the status message as well.

Reply:
nice fix. I think I would prefer if `_nextForkBlock` was null or undefined rather than zero for this check if there is no next fork coming, but that is a small nit. I'm not exactly sure what other repercussions that may have in the code base, `0` could very much be the better choice other than this buffer conversion issue.